lib/dgram.js

```diff
@@ -78,7 +78,7 @@ exports._createSocketHandle = function(address, port, addressType, fd, flags) {
 function Socket(type, listener) {
   EventEmitter.call(this);
 
-  if (typeof type === 'object') {
+  if (type !== null && typeof type === 'object') {
```

Comment:
@claudiorodriguez This does indeed seem to be different with `!== null`. You can feel free to disregard my comment :)
Expressions of the form `a && typeof a === 'object'` definitely seem to be a reason for deopts, though.